splink/blocking.py

```diff
@@ -48,6 +53,60 @@ def salted_blocking_rules(self):
             for n in range(self.salting_partitions):
                 yield f"{self.blocking_rule} and ceiling(l.__splink_salt * {self.salting_partitions}) = {n+1}"  # noqa: E501
 
+    @property
+    def _parsed_join_condition(self):
+        br = self.blocking_rule
+        return parse_one("INNER JOIN r", into=Join).on(
+            br, dialect=self.sqlglot_dialect
+        )  # using sqlglot==11.4.1
```

Comment:
I'll start a formal review shortly, but can you sort out the poetry dependencies if sqlglot 11.4.1 is required for this PR?

I would imagine we can pin the version to 11.4.1 without any issues.

Reply:
Good point. I haven't tried any other version actually, I just was m aking a note for any future readers of the code.  Definitely worth checking a couple of other versions in the range currently specified

Reply:
[Done](https://github.com/moj-analytical-services/splink/pull/1388/commits/4b992da1f9d9ad3ab663dac91389a8fdfc25d535)